testing/regression/CMakeLists.txt

```diff
@@ -1,3 +1,7 @@
+if ("thrust" IN_LIST THRUST_TARGETS)
```

Comment:
We'll need to move this test into `testing/` and shorten its name (probably just `iterator_categories.cu`). The tests in the `testing/regression` directory are problematic and need to be cleaned up. See #1205 for details.